src/System.Private.ServiceModel/tests/Scenarios/Contract/Fault/FaultExceptionTests.4.0.0.cs

```diff
@@ -299,4 +301,74 @@ public static void FaultException_FaultContractAndKnownType_Echo()
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
         }
     }
+
+#if FULLXUNIT_NOTSUPPORTED
+    [Fact]
+#endif
+    [WcfFact]
+    [OuterLoop]
+    public static void TestAccessFaultContractInfo()
+    {
+        MyClientBase<IWcfService> client = null;
+        IWcfService serviceProxy = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding customBinding = new CustomBinding();
+            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+            customBinding.Elements.Add(new HttpTransportBindingElement());
+
+            client = new MyClientBase<IWcfService>(customBinding, new EndpointAddress(Endpoints.DefaultCustomHttp_Address));
+            client.Endpoint.EndpointBehaviors.Add(new TestFaultContractInfosBehavior());
+            serviceProxy = client.ChannelFactory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            var input = new object[] { new FaultDetail(), new KnownTypeA() };
+            // Use exiting service as we only need to verify FaultContractInfo can be accessed from ClientOperation
+            serviceProxy.TestFaultWithKnownType(null, input);
```

Comment:
I think it would be better to do the testing of the FaultContractInfo in the ClientOperations here, not inside the Behavior.  You could make the Behavior just capture the ClientOperations into a property you can interrogate from this test. I think it should be obvious just by reading any test what is being tested.  As written here, you have to poke around in other code to find out what is actually being verified.